File: mooc_application/src/main/java/com/hongxuan/mooc/service/FileResourcesService.java
package com.hongxuan.mooc.service;
import com.hongxuan.pojo.TbFileResources;
import entity.PageResult;

import java.util.List;
/**
 * 业务逻辑接口
 * @author LiaoHongXuan
 *
 */
public interface FileResourcesService {

	/**
	 * 返回全部列表
	 * @return
	 */
	public List<TbFileResources> findAll();
	
	
	/**
	 * 返回分页列表
	 * @return
	 */
	public PageResult findPage(int pageNum,int pageSize);
	
	
	/**
	 * 增加
	*/
	public void add(TbFileResources fileResources);
	
	
	/**
	 * 修改
	 */
	public void update(TbFileResources fileResources);
	

	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
	public TbFileResources findOne(Long id);
	
	
	/**
	 * 批量删除
	 * @param ids
	 */
	public void delete(Long [] ids);

	/**
	 * 分页
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(TbFileResources fileResources, int pageNum,int pageSize);
	
}

File: mooc_application/src/main/webapp/js/service/fileResourcesService.js
//服务层
app.service('fileResourcesService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('../fileResources/findAll.do');		
	}
	//分页 
	this.findPage=function(page,rows){
		return $http.get('../fileResources/findPage.do?page='+page+'&rows='+rows);
	}
	//查询实体
	this.findOne=function(id){
		return $http.get('../fileResources/findOne.do?id='+id);
	}
	//增加 
	this.add=function(entity){
		return  $http.post('../fileResources/add.do',entity );
	}
	//修改 
	this.update=function(entity){
		return  $http.post('../fileResources/update.do',entity );
	}
	//删除
	this.dele=function(ids){
		return $http.get('../fileResources/delete.do?ids='+ids);
	}
	//搜索
	this.search=function(page,rows,searchEntity){
		return $http.post('../fileResources/search.do?page='+page+"&rows="+rows, searchEntity);
	}    	
});
